packages/@react-aria/gridlist/src/useGridList.ts

```diff
@@ -51,6 +51,11 @@ export interface AriaGridListProps<T> extends GridListProps<T>, DOMProps, AriaLa
 export interface AriaGridListOptions<T> extends Omit<AriaGridListProps<T>, 'children'> {
   /** Whether the list uses virtual scrolling. */
   isVirtualized?: boolean,
+  /**
+   * Whether typeahead navigation is disabled.
+   * @default false
+   */
+  disallowTypeAhead?: boolean,
```

Comment:
With the eventual changes to stop propagation on handled keyboard events as mentioned in https://github.com/adobe/react-spectrum/pull/7376#issuecomment-2524214331, hopefully we would be able to make the typeahead listener not capturing which would help prevent typeahead from colliding with nested component/grid interactions. We would probably need to special case `Space` so as to not collide with row selection and type ahead, but that might be doable via checking if the `Space` in the middle of a string the user is typing (aka that would be typeahead so don't perform row selection) or not (aka that would be row selection so stop propagation at the row level). This would mean communicating from the grid level to the row level that typeahead is in progress or not though

If that was accomplished though, would you need `disallowTypeAhead` anymore?

Reply:
Hm, sounds like that will take a while, but yes I think that could work 👍 

Unfortunately, I think `Space` will not be the only special case to handle. I also see potential collisions with alphanumeric cell entry once the full W3C grid edit mode spec is adopted.

Reply:
> Maybe disallowTypeAhead could still be passed undocumented in the meantime?

potentially if only to get something out for you sooner rather than later, I'll ask the team

Reply:
so the team is ok with exposing `disallowTypeAhead` since it isn't required behavior as per the grid spec and there is still the potential of collisions as you mentioned above. As for the collection id stuff that we chatted above above, are those changes required or could you ensure that the ids for each collection are globally unique (though I guess you can't if users are the ones providing those)?

Reply:
Great 👍

In terms of the collection ids, yes those changes are required for the reason you mentioned.

How would the team view a removal of the `id` property from leaf components?

Ofc it could impact existing applications if they are relying on the`data-key` for querying/styling, but this would allow us to streamline callback behavior and solve the outstanding issue here at once. Impacted users would have to migrate to a custom data attribute.

Im out of ideas on how else to approach this issue 😬

Reply:
I did want to maintain the extraction performed by `useCachedChildren` and only remove the `rendered.props.id` part of it. Was thinking we might be able to get away with using auto-incremental ids to maintain the ability to reverse map. Not a big fan of this myself however because of re-ordering. 

In the meantime, I thought of another possible angle which i pushed up. Let me know what you think 👍

Reply:
Thanks for the update! I think this is promising, we may need to think about moving the id unscoping to the hook level perhaps in case there are React Aria hook only users who've also leveraged the Collection implementation. I also noticed that the `idScope` isn't being applied for static collections that don't have `id`'s provided to their children (in your story the GridListItems still have `react-aria-1`/etc instead of being scoped), I assume that wasn't intentional? 
The condition below that you have in `useCachedChildren` will return true since the child doesn't have an id, perhaps it include that in the check?
```
if (!child || !child.props.id || !idScope) {return child;}
```

Reply:
I added that intentionally for now until I figure out what causes the `Table.test.js` to fail when not early-returning. Unfortunately I only get an undescriptive `Should not already be working.` error message.

Also wanted to note that key structure will change slightly for these static nodes, since I don't have access to the auto-generated `node.key` at this point in time. Instead I think the index (e.g. `<idScope>:<index>`) will have to do.

I also thought about moving the unscoping into the hook layer, but wasn't sure whether we actually would want that. Since `idScope` is pretty implementation specific & set in the component layer I figured coupling it there might make more sense. I also felt like `useListState()` shouldn't have to be aware of a scope.

In case it shall still move further down, we would have to either have forward the `idScope` everywhere or maybe attach it to the `BaseCollection`, since we can't access the `CollectionContext` in the hidden tree unfortunately. 

Looking forward to additional feedback here 👍

Reply:
Btw, another question popped up while fixing `Table.test.js`. Maybe its worth thinking about merging these together instead of picking either one (e.g. `<ctxIdScope>:<idScope>:<index>`).

https://github.com/adobe/react-spectrum/blob/fee532d6a2fc025177d96823d0b5e361b4b79934/packages/%40react-aria/collections/src/CollectionBuilder.tsx#L202

Reply:
> Also wanted to note that key structure will change slightly for these static nodes, since I don't have access to the auto-generated node.key at this point in time. Instead I think the index (e.g. <idScope>:<index>) will have to do.

I think that is ok, I feel like the end user mainly cares about the structure/format of the node key w/ respect with what gets returned by the various event handlers. I personally don't consider the `data-key` as public API but will discuss with the team further.

> I also thought about moving the unscoping into the hook layer, but wasn't sure whether we actually would want that. 
>  Since idScope is pretty implementation specific & set in the component layer I figured coupling it there might make more sense. I also felt like useListState() shouldn't have to be aware of a scope.
In case it shall still move further down, we would have to either forward the idScope or maybe attach it to the BaseCollection, since we can't access the CollectionContext in the hidden tree unfortunately.

That's true, mainly was thinking if hook users would want to implement nested collection support themselves and/or support it out of the box but that can be addressed down the line. 

> Maybe its worth thinking about merging these together instead of picking either one (e.g. <ctxIdScope>:<idScope>:<index>).

Interesting, so that would essentially append the parent collection's scope to the child collection, automatically ensuring each level would have a unique scopeId even if the user provided the same idScope to both the parent and the child. I think that is probably worthwhile, digging to see if there would any reason why not.